File: abstract-factory/php/OracleDatabaseFactory.php
<?php

namespace AbstractFactory;

use AbstractFactory\IDatabaseFactory;
use AbstractFactory\Connection;
use AbstractFactory\Command;
use AbstractFactory\OracleConnection;
use AbstractFactory\OracleCommand;

class OracleDatabaseFactory implements IDatabaseFactory
{
    function createConnection(): Connection
    {
        return new OracleConnection();
    }

    function createCommand(): Command
    {
        return new OracleCommand();
    }
}

File: abstract-factory/dart/oracle_database_factory.dart
import 'command.dart';
import 'connection.dart';
import 'i_database_factory.dart';
import 'oracle_command.dart';
import 'oracle_connection.dart';

final class OracleDatabaseFactory implements IDatabaseFactory {
  const OracleDatabaseFactory();

  @override
  Command createCommand() {
    return const OracleCommand();
  }

  @override
  Connection createConnection() {
    return const OracleConnection();
  }
}
